File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
//         map< int , int> mp;
//         for(int i = 0; i < nums.size(); i++)
//         {
//             mp[nums[i]]++;
//         }
//         for(int i = 0; i < nums.size(); i++)
//         {
//             if(mp[nums[i]] > (nums.size()/2))return nums[i];
            
//         }
        sort(nums.begin(),nums.end());
        int mid = nums.size() / 2;
        
        return nums[mid];
    }
};

File: 169-majority-element/169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let mp = new Map();
    for(let i = 0; i < nums.length; i++){
        let count = mp.get(nums[i]) || 0;
        mp.set(nums[i],++count);
    }
    for(let i = 0; i < nums.length; i++){
        if(mp.get(nums[i]) > (nums.length/2))return nums[i];
    }
    return 0;
};